contracts/pToken.sol

```diff
@@ -168,4 +171,18 @@ contract PToken is
         ORIGIN_CHAIN_ID = _newOriginChainId;
         return true;
     }
+
+    function addToTokensReceivedWhitelist(address _address) public onlyAdmin {
```

Comment:
if this function is only called externally I would use `external` in order to save a bit of gas.

Reply:
Oops yeah good spot - done!